From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alexandra-Myers <bia416cat@gmail.com>
Date: Tue, 23 Sep 2025 21:10:20 -0400
Subject: [PATCH] Use legacy tracker ticking

Original license: GPL-3.0-only
Original project: https://github.com/PaperMC/Paper
Paper pull request: https://github.com/PaperMC/Paper/pull/13094

Related MC issue: https://mojira.dev/MC-297196

diff --git a/net/minecraft/server/level/ServerChunkCache.java b/net/minecraft/server/level/ServerChunkCache.java
index dcef83f4b01bfaa2a755f55d0b760af9535be742..608c92e1544801bd080be9107cdb39af98da6416 100644
--- a/net/minecraft/server/level/ServerChunkCache.java
+++ b/net/minecraft/server/level/ServerChunkCache.java
@@ -516,7 +516,7 @@ public class ServerChunkCache extends ChunkSource implements ca.spottedleaf.moon
         if (tickChunks) {
             ((ca.spottedleaf.moonrise.patches.chunk_system.level.ChunkSystemServerLevel)this.level).moonrise$getPlayerChunkLoader().tick(); // Paper - rewrite chunk system
             this.tickChunks();
-            this.chunkMap.tick();
+            if (!org.dreeam.leaf.config.modules.gameplay.Knockback.useLegacyTrackerTicking) this.chunkMap.tick(); // Leaf - Use legacy tracker ticking
         }
 
         profiler.popPush("unload");
diff --git a/net/minecraft/server/level/ServerLevel.java b/net/minecraft/server/level/ServerLevel.java
index 8478f2e66c8a9fd78996b98700006471ccc926ef..c22c4bb8bf458813848ab1d6f362cdec076af1e0 100644
--- a/net/minecraft/server/level/ServerLevel.java
+++ b/net/minecraft/server/level/ServerLevel.java
@@ -997,6 +997,7 @@ public class ServerLevel extends Level implements WorldGenLevel, ServerEntityGet
         profiler.push("entityManagement");
         // Paper - rewrite chunk system
         profiler.pop();
+        if (org.dreeam.leaf.config.modules.gameplay.Knockback.useLegacyTrackerTicking) this.getChunkSource().chunkMap.tick(); // Leaf - Use legacy tracker ticking
         profiler.push("debugSynchronizers");
         if (this.debugSynchronizers.hasAnySubscriberFor(DebugSubscriptions.NEIGHBOR_UPDATES)) {
             this.neighborUpdater
